### ISSUE TITLE
[NEP issues] Absolute value of DFT energy may influence the GPU accuracy

### ISSUE THREAD

**Issue by neetsaki** [brucefan1983/GPUMD#258]
In the CP2k Program, the abolute energy of large systems may exceed 290000. The implement of GPUMD use the **single precision** and thus operate the float data with 8 significant figures. Most differentiation in large number (such as the difference between -291230.1 and -291241.1) will be ignored. Hence, the NEP program can only learn the limited features of train set.
For current version, it's adviced that user subtract an comparative energy for all configuration in the train set to make the differnce between data more distinct. (291230.1/-291241.1  vs  30.1/41.1) 

(When I try subtract them by the minimum energy of train set, the loss function of energy decrease very slowly and fluctuate  intensely (as the tail of red curve shown). This process need longer training time to valid, and I'm not sure if this solution works now)

<img width="1179" alt="image" src="https://user-images.githubusercontent.com/40904479/188350666-8b50c5f5-f569-44c5-9ff9-3d24a86d0111.png">

**Comment by brucefan1983** [comment]
Too large absolute energy values indeed can affect the training accuracy to some degree, but your current training data seem to have bigger problems than this. So
* I will consider energy shift and normalization in the future, but
* I suggest you check your training data carefully.

**Comment by neetsaki** [comment]
Thanks for your advice. I check the ditribution of force and find it appears some bugs (miss all force value miss +/- symbol) and the results become much better. 
<img width="1175" alt="image" src="https://user-images.githubusercontent.com/40904479/188394065-64109fe0-db27-461f-82a8-dc29b89f9c92.png">

**Comment by brucefan1983** [comment]
Looks much more reasonable now, but I hope the accuracy can be improved further :-)

**Comment by brucefan1983** [comment]
I think this is not an issue any more.
